backend/lcfs/web/api/initiative_agreement/validation.py

```diff
@@ -40,3 +43,27 @@ async def validate_initiative_agreement_update(
                 status_code=403,
                 detail="Editing an approved initiative agreement is not allowed.",
             )
+
+    async def validate_organization_access(self, initiative_agreement_id: int):
+        initiative_agreement = await self.service.get_initiative_agreement(initiative_agreement_id)
+        if not initiative_agreement:
+            raise HTTPException(
+                status_code=status.HTTP_404_NOT_FOUND,
+                detail="Transaction not found.",
+            )
+
+        organization_id = initiative_agreement.to_organization.organization_id
+        user_organization_id = (
```

Comment:
suggestion: This block is repeated a few times, does it make sense to DRY this up?

Reply:
yes, since the structure is different for each of the feature type, had to do this way. But will look if I can generalize this.